### ISSUE TITLE
ngcc process fails with vscode versions 1.62.1 and above

### ISSUE THREAD

**Issue by joeypedicini92** [angular/vscode-ng-language-service#1565]
# 🐞 bug report

### Is this a regression?

 Yes, I am currently on v13.0.0, I don't remember when it started throwing this error.

### Description

 A clear and concise description of the problem...

I'm getting a popup everytime tsconfig.json changes and angular language service starts. Happening in multiple Angular v12 repos

## Bug Type
What does this bug affect

- [x] Angular Language Service VSCode extension
- [ ] Angular Language Service server

## Reproduction

**Expected behavior**

### **Logs**

Log file gives us deep insight into the behavior and performance of the extension. If the issue is a performance
problem or an error occured, please provide the output of the log file below.

Set Angular Log level to verbose.

<pre><code>
[Trace - 9:44:04 AM] Sending notification 'workspace/didChangeWatchedFiles'.
[Trace - 9:44:04 AM] Received notification 'angular/projectLoadingFinish'.
[Trace - 9:44:04 AM] Received notification '$/progress'.
[Trace - 9:44:05 AM] Received notification 'window/logMessage'.
[Error - 9:44:05 AM] Failed to run ngcc for /Users/joey/Repos/battleface-v2/tsconfig.json, language service may not operate correctly:
    ngcc for /Users/joey/Repos/battleface-v2/tsconfig.json returned exit code null, stderr: 
[Trace - 9:44:05 AM] Received notification '$/progress'.
[Trace - 9:44:05 AM] Received notification 'window/logMessage'.
[Info  - 9:44:05 AM] Enabling Ivy language service for /Users/joey/Repos/battleface-v2/tsconfig.json.

</code></pre>

### **Screenshots**

<img width="452" alt="Screen Shot 2021-11-11 at 9 44 11 AM" src="https://user-images.githubusercontent.com/6809064/141317933-68b1aa88-204f-4c2c-ab76-dd643618fa85.png">

## 🌍  Your Environment
**Angular Version:**
<pre><code>

 12.2.7

</code></pre>

Extension Version:
<pre><code>

 v13.0.0

</code></pre>

VSCode Version:
<pre><code>

 Version: 1.62.1 (Universal)

</code></pre>

Operating System:
<pre><code>

 macOS 11.6

</code></pre>

**Anything else relevant?**

**Comment by atscott** [comment]
@joeypedicini92 a few questions to help, as I have not been able to reproduce this myself:

* Do you have an example project and what are the steps to reproduce with an example project?
* Are your v12 projects using Ivy or View Engine? (is `"enableIvy": false` in the `tsconfig.json`)
* Are the dependencies installed for the project? (`yarn/npm install`)
* Does `ngc` fail when run manually on the project? (`./node_modules/.bin/ngc tsconfig.json`)
* Does `ngcc` fail when run manually on the project? (`./node_modules/.bin/ngcc --tsconfig tsconfig.json`)

**Comment by joeypedicini92** [comment]
I can try to get a sample project together when I have a bit more time. But it happens as soon as I open a project in vscode.

I am using Ivy, however in the vscode angular language server settings if I disable Ivy, I no longer get these errors (but then type checking and everything seems to not work right)

dependencies are installed, using yarn

ngc worked

ngcc worked

for additional reference I'm using an Nx repo in one repo but the other is just a plain angular repo

**Comment by smlombardi** [comment]
This is happening to me as well, as of the Insider version release 11/10. Same specs.  We have multiple projects in our repo, with a tsconfig.json at the root of the repo:
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "src",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "module": "esnext",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es2015",
    "typeRoots": ["node_modules/@types"],
    "lib": ["es2018", "dom"],
    "paths": {
      "@app/*": ["app/*"],
      "@shared/*": ["app/shared/*"],
      "@env/*": ["environments/*"],
      "pb-design-system": ["../dist/pb-design-system"],
      "pb-design-system/*": ["../dist/pb-design-system/*"]
    }
  }
}
```

and then in the main project where Angular is (and all the files), `/client`, we have tsconfig.app.json:
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": ""
  },
  "files": [
    "main.ts",
    "polyfills.ts"
  ],
  "include": [
    "src/**/*.d.ts"
  ]
}
```

<img width="941" alt="clientsrcappsharedbadgesaccessibility-badge component ts — ng-designsystem-20211111_131531" src="https://user-images.githubusercontent.com/835148/141349041-b13c29e4-dad9-4597-9714-18082f23e57a.png">

---

I am also getting this error on all our other projects, which use Nx Workscapes. These consist of a series of libs, each of which has it's own tsconfig.json file and they all point to the main one at the root:
```
{
  "extends": "../../../tsconfig.base.json",
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ],
  "angularCompilerOptions": {
    "strictInjectionParameters": true,
    "strictTemplates": true
  }
}
```

Not sure if any of this matters, since it all worked fine like this up until 2 days ago when the error occurred.

**Comment by atscott** [comment]
@smlombardi You mentioned this was working for you two days ago. What updates happened between then and now? If nobody can provide a reproduction, hopefully we can at least determine what version this started happening with. Can you try installing different versions of the language service extension to see if there is an older one that works? If not, then the issue may have been caused by an update to the libraries in your project and it would be good to find what update that was.

**Comment by smlombardi** [comment]
It happened after I updated Code to the insiders version that came out 2 days ago. It happened immediately on relaunch

**Comment by atscott** [comment]
@smlombardi Just to clarify, there were no other updates to the project or extension? Updating the Code version didn't result in an update to the vscode extension? I'll go grab the latest insider version now to see if I can reproduce this.

**Comment by atscott** [comment]
Okay, I have been able to isolate this to the vscode `1.62.1` release. The error does not occur in the `1.62.0` release. When updating to `1.62.1` with no other changes, the `ngcc` process fails.

**Comment by atscott** [comment]
We are still actively investigating this issue and are tracking https://github.com/microsoft/vscode/issues/136988 as the likely resolution. 

In the meantime, the error we show does give the appropriate information. `ngcc` failed to run, which _might_ cause the language service to not work correctly but can be resolved by `npm/yarn run ngcc`. Or if you are already running a dev server, `ngcc` likely already ran and you don't have to do anything.

**Comment by atscott** [comment]
Closing as resolved in the `1.62.3` release of VSCode.
